### ISSUE TITLE
Tests failing on python 3.6

### ISSUE THREAD

**Issue by npmccallum** [latchset/kdcproxy#24]
https://koji.fedoraproject.org/koji/taskinfo?taskID=17158477

**Comment by tiran** [comment]
It's not just Python 3.6. 2.7 is failing, too.

**Comment by tiran** [comment]
Python 2.7 breakage is caused by https://github.com/testing-cabal/mock/issues/323

```
>>> mock.patch('socket.socket', autospec=True).__enter__()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "mock/mock.py", line 1447, in __enter__
    _name=self.attribute, **kwargs)
  File "mock/mock.py", line 2328, in create_autospec
    _name='()', _parent=mock)
  File "mock/mock.py", line 2367, in create_autospec
    _check_signature(original, new, skipfirst=skipfirst)
  File "mock/mock.py", line 207, in _check_signature
    _copy_func_details(func, checksig)
  File "mock/mock.py", line 233, in _copy_func_details
    funcopy.func_defaults = func.func_defaults
AttributeError: 'functools.partial' object has no attribute 'func_defaults'
```

Python 3.6 breakage seems to be caused by a more recent version of pytest. It considers webtest.TestApp as test class but then refuses to use it because it has a ```__init__ ``` method.

**Comment by npmccallum** [comment]
Fixed in d9923b582a31cafce75ddd3a97857fa1794e22ca
